src/analyses/base.ml

```diff
@@ -1637,28 +1637,93 @@ struct
     in
     let inv_bin_float (a, b) fkind c op =
       let meet_bin a' b'  = FD.meet a a', FD.meet b b' in
-      let meet_com oi = (* commutative *)
-        meet_bin (oi c b) (oi c a) in
-      let meet_non oi oo = (* non-commutative *)
-        meet_bin (oi c b) (oo a c) in
       match op with
-      | PlusA  -> meet_com FD.sub
+      | PlusA  ->
+        (* A + B = C, \forall a \in A. a + b_min > pred c_min \land a + b_max < succ c_max
+            \land a + b_max > pred c_min \land a + b_min < succ c_max
+           \rightarrow A = [min(pred c_min - b_min, pred c_min - b_max), max(succ c_max - b_max, succ c_max - b_min)]
```

Comment:
Why isnt it enought to say `A = [pred c_min - b_max, succ c_max - b_min]`?
By definition `b_min <= b_max`, and therefore `c_min - b_min <= c_min - b_max`. so simpler would be `min(pred c_min - b_min, pred c_min - b_max) = pred c_min - b_max` (similar for upper bound)

Reply:
With `b_min = -100` and `b_max = 0`, `c_min - b_min` is not necessary smaller then `c_min - b_max`. Maybe there are some simplifications possible, however i prefer to not apply them here as it is very easy to get it wrong